src/lib/autoinstall/autosetup_helpers.rb

```diff
@@ -98,12 +97,60 @@ def suse_register
 
         # failed relnotes download is not fatal, ignore ret code
         Yast::WFM.CallFunction("inst_download_release_notes")
-      elsif general_section["semi-automatic"]&.include?("scc")
+      elsif semi_auto?("scc")
         Yast::WFM.CallFunction("inst_scc", ["enable_next" => true])
       end
       true
     end
 
+    # Convenience method to fetch the general profile section
+    #
+    # @return [Hash] the general section from the current profile when present;
+    #   an empty hash when not
+    def general_section
+      Yast::Profile.current["general"] || {}
+    end
+
+    # Convenienve method to check whether a particular client should be run to
+    # be configured manually during the autoinstallation according to the
+    # semi-automatic section
+    #
+    # @return [Boolean]
+    def semi_auto?(name)
+      !!general_section["semi-automatic"]&.include?(name)
+    end
+
+    # Autosetup the network
+    def autosetup_network
+      if Yast::Profile.current["networking"]
+        if Yast::Profile.current["networking"]["setup_before_proposal"]
+          log.info("Networking setup before the proposal")
+          Yast::AutoinstConfig.network_before_proposal = true
```

Comment:
Is the `#network_before_proposal` method used elsewhere? If that's the case, just keep it as it is. If it is not, I guess we could get rid of it (and use local variable).

Reply:
I will check, it was exposed in the past because could be needed by other modules during the autoconfiguration workflow, like for example save_network.

That information is already stored in Lan when the import is done, so probably we could get rid of it. 

But maybe we could do it in a second round. So, finishing with the yast2-network changes related to this first stage configuration first and then deciding where to keep that responsibility.

What do you think?

Reply:
It was introduced by me in: https://github.com/yast/yast-autoinstallation/pull/347

Reply:
OK. Please, make a note somewhere :) and let's continue.